src/lib/components/AccessRightField.js

```diff
@@ -49,37 +47,49 @@ class Protection {
 }
 
 
-export class AccessRightField extends Component {
+function convertToNewFormat(access) {
+  let record;
+  if (access.record && ["public", "restricted"].includes(access.record)) {
+    record = access.record;
+  } else {
+    record = access.metadata ? "restricted" : "public";
+  }
+  let files;
+  if (access.files && ["public", "restricted"].includes(access.files)) {
+    files = access.files;
+  } else {
+    files = access.files ? "restricted" : "public";
+  }
+
+  return {
+    record,
+    files,
+    owned_by: access.owned_by,
+    embargo: access.embargo ? access.embargo : {
+      "active": false,
+      "until": "",
+      "reason": ""
+    },
+    grants: access.grants ? access.grants : []
+  }
+}
+
+export class AccessRightFieldComponent extends Component {
   /** Top-level Access Right Component */
 
   render() {
-    const { fieldPath, label, labelIcon, options } = this.props;
-
-    // External
-    // TODO: replace by Formik access field slice
-    const tmpMetadataPublic = false;
-    const tmpHasFiles = true;
-    const tmpFilesPublic = false;
-    const tmpisEmbargoApplied = true;
-    const tmpEmbargoDate = "2021-06-21";
-    const tmpEmbargoLifted = false;
-    const tmpEmbargoReason = "";
-
-
-    const embargoState = EmbargoState.from(
-      tmpMetadataPublic,
-      tmpFilesPublic,
-      tmpisEmbargoApplied,
-      tmpEmbargoLifted
-    );
-
-    const protection = Protection.create(
-      // TODO: replace by Formik access field slice
-      tmpMetadataPublic,
-      tmpFilesPublic,
-      tmpHasFiles,
-      new Embargo({state: embargoState, date: tmpEmbargoDate, reason: tmpEmbargoReason})
-    );
+    const {
+      fieldPath,
+      formik,  // this is our access to the shared current draft
+      hasFiles,
+      label,
+      labelIcon,
+    } = this.props;
+
+    // value of access field
+    // temporarily convert to upcoming backend format
+    const access = convertToNewFormat(formik.field.value);
```

Comment:
I see the pattern you use here with the `formik.field.value` and I am wondering if we should use it in the other components to avoid `getIn(formik.form.values, fieldPath, defaultValue)` or stick to the old way for consistency..... My main concern is that we are starting to create different habits of doing things and codebase is getting messy.... Maybe we create an issue to evaluate that?

Reply:
:grin: Ha! This is going away, so it should be fine. The other place using `formik.field.value` is in utils and it's for the reason you mentioned: saves us the trouble of trying to find the field value ourselves when Formik already gives it to us.

About the consistency, I think I agree with tending towards consistency long-term and being self-consistent within a PR. But I don't think I agree with consistency short term. In order to improve things, we need to be able to "run experiments" and do things differently if we think it's better. However it's hard to know if some ways are better right away, so retrofitting immediately in the name of consistency lends itself to "rehacktoring" (churn in place). 

Unfortunately, I don't know how to operationalize this: my experience has been that "refactoring" tickets don't work (nobody takes them because they are not "needed" - 'hey it works, don't touch it'). The boy scout rules seems to work better, but when do we know/communicate that a pattern is good enough to be spread around? Seems like an organic process in the end. For example, I think the pattern you've introduced for Formik wrapper:

```js
      <Field
        name={this.props.fieldPath}
        component={(formikProps) => ( 
          <WrappedComponent formik={formikProps} {...this.props} />
        )}
      />
```
is really nice because of readability and clear separation and I am re-using it now.